Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -89,6 +89,59 @@ namespace AZ::Dom
         return !operator==(key);
     }
 
+    bool PathEntry::operator<(const PathEntry& other) const
+    {
+        const size_t* otherIndexValue = AZStd::get_if<size_t>(&other.m_value);
+        const AZ::Name* otherNameValue = AZStd::get_if<AZ::Name>(&other.m_value);
+        if (otherIndexValue)
+        {
+            return operator<(*otherIndexValue);
+        }
+        return operator<(*otherNameValue);
+    }
+
+    bool PathEntry::operator<(size_t index) const
+    {
+        const size_t* internalValue = AZStd::get_if<size_t>(&m_value);
+        return internalValue == nullptr || *internalValue < index;
+    }
+
+    bool PathEntry::operator<(const AZ::Name& key) const
+    {
+        return operator<(key.GetStringView());
+    }
+
+    bool PathEntry::operator<(AZStd::string_view key) const
+    {
+        const AZ::Name* internalValue = AZStd::get_if<AZ::Name>(&m_value);
+        return internalValue != nullptr &&
+            AZStd::lexicographical_compare(
+                   internalValue->GetStringView().begin(),
+                   internalValue->GetStringView().end(),
+                   key.begin(),
+                   key.end());
+    }
+
+    bool PathEntry::operator>(const PathEntry& other) const
```

Comment:
couple things:
1) generally containers just use operator< for comparisons, so the operator> is largely unnecessary
2) the inverse of lessThan isn't greaterThan, it's greaterThanOrEqualTo, so this code is incorrect. Honestly, I would just remove these operator> altogether

Reply:
Makes sense. I'll remove the operator>. We can revisit it if needed later on.

Reply:
If you are going to implement the less than operator, than either it should implemented by itself or all four of the comparison operators should be implemented as well(operator<, operator>, operator<=, operator>=).

In no situation should only 2 or 3 out of the 4 comparison be implemented.
It would surprising to users of the class if they had `operator>` but not `operator<`= or `operator>=`

Reply:
Also all the other operators should be implemented in terms of operator<
```
bool PathEntry::operator>(const PathEntry& other) const
{
    return other < *this;
}
bool PathEntry::operator<=(const PathEntry& other) const
{
    return !(other < *this);
}
bool PathEntry::operator>=(const PathEntry& other) const
{
    return *this < other;
}
```

Reply:
Working on adding the other operators (<=, >=) and fixing up some other issues I noticed now. Will look at your suggestions above and get another commit up.

Reply:
After looking into what it would take to add (>, <=, >=) to PathEntry, I decided to leave the class as is with operator< only. If the need arises for the others then we can implement them.